internal/k8s/exec.go

```diff
@@ -28,12 +30,25 @@ func (k *K8sClient) Exec(ctx context.Context, podID PodID, cName container.Name,
 
 	exec, err := remotecommand.NewSPDYExecutor(k.restConfig, "POST", req.URL())
 	if err != nil {
-		return err
+		return fmt.Errorf("failed to establish connection: %v", err)
```

Comment:
fwiw, i would probably shorted this to "establishing connection: %v", on the assumption that callers may wrap this in their  own error message (and so we don't need to say "failure" each time)